components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +263,86 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
-		if proxy.Name == "cluster" {
-			if proxy.Status.HTTPProxy != "" && proxy.Status.HTTPSProxy != "" &&
-				proxy.Status.NoProxy != "" {
-				// Update Proxy Env variables map
-				proxyEnvVars["HTTP_PROXY"] = proxy.Status.HTTPProxy
-				proxyEnvVars["HTTPS_PROXY"] = proxy.Status.HTTPSProxy
-				proxyEnvVars["NO_PROXY"] = proxy.Status.NoProxy
-				if proxy.Spec.TrustedCA.Name != "" {
-					proxyEnvVars["PIP_CERT"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["REQUESTS_CA_BUNDLE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["SSL_CERT_FILE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-				}
-				return true
-			}
-		}
-	}
-	return false
-
-}
-
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.Decoder = d
 	return nil
 }
 
-func InjectProxyConfig(notebook *nbv1.Notebook) error {
-	notebookContainers := &notebook.Spec.Template.Spec.Containers
-	var imgContainer corev1.Container
+func CheckAndMountCACertBundle(ctx context.Context, cli client.Client, notebook *nbv1.Notebook) error {
 
-	// Add trusted-ca volume
-	notebookVolumes := &notebook.Spec.Template.Spec.Volumes
-	certVolumeExists := false
-	certVolume := corev1.Volume{
-		Name: "trusted-ca",
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "trusted-ca",
-				},
-				Items: []corev1.KeyToPath{{
-					Key:  "ca-bundle.crt",
-					Path: "tls-ca-bundle.pem",
-				},
-				},
-			},
-		},
-	}
-	for index, volume := range *notebookVolumes {
-		if volume.Name == "trusted-ca" {
-			(*notebookVolumes)[index] = certVolume
-			certVolumeExists = true
-			break
-		}
-	}
-	if !certVolumeExists {
-		*notebookVolumes = append(*notebookVolumes, certVolume)
+	configMapName := "odh-trusted-ca-bundle"
+
+	// Fetch the list of ConfigMaps from the cluster
+	configMapList := &corev1.ConfigMapList{}
+	if err := cli.List(ctx, configMapList); err != nil {
+		return err
 	}
 
-	// Update Notebook Image container with env variables and Volume Mounts
-	for _, container := range *notebookContainers {
-		// Update notebook image container with env Variables
-		if container.Name == notebook.Name {
-			var newVars []corev1.EnvVar
-			imgContainer = container
-
-			for key, val := range proxyEnvVars {
-				keyExists := false
-				for _, env := range imgContainer.Env {
-					if key == env.Name {
-						keyExists = true
-						// Update if Proxy spec is updated
-						if env.Value != val {
-							env.Value = val
-						}
-					}
-				}
-				if !keyExists {
-					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
-				}
+	// Search for the odh-trusted-ca-bundle ConfigMap
+	for i := range configMapList.Items {
+		cm := &configMapList.Items[i]
+		if cm.Name == configMapName {
+
+			// Check if the odh-ca-bundle.crt key exists
+			certData, ok := cm.Data["odh-ca-bundle.crt"]
```

Comment:
As we are checking for certs, 
we should check for both, the configmap currently as two, we should check for both.

Reply:
Added the same check for `ca-bundle.crt`, but since it has the ability to auto-heal, I couldn't test it with wrong certificate.